### DISCUSSION TITLE
Property 'PrismaClientKnownRequestError' does not exist on type 'typeof Prisma'.

### DISCUSSION THREAD

**Discussion by pfiadDi** [prisma/prisma#17478] [Q&A]
Hi, I get above error when I try to to check the error type as suggested in the docs:

`    if (err instanceof Prisma.PrismaClientKnownRequestError) {`

I import: 

`import { Prisma } from "@prisma/client";`

Do I have to install additional types (none found) or do I need to import anything else?

Thanks
Br

**Comment by nurul3101** [comment]
Hello @pfiadDi 👋 

No, you don't need to import any additional types.
Have you invoked `npx prisma generate` to generate all the types based on your schema file?
Can you try invoking the generate command and check if you get the same error?

**Reply by pfiadDi** [reply]
Hi @nurul3101  thank you for your answer.
I just run it again and still the same error.

**Reply by nurul3101** [reply]
Is your project open source?
If yes, can you share the link so that I can check what the issue could be?

**Comment by horlaarsco** [comment]
I am getting this same error. Was this ever resolved?

**Comment by roberta-mitiga** [comment]
Same error here:

**Property 'PrismaClientKnownRequestError' does not exist on type 'typeof Prisma**

    import` { Prisma } from '@prisma/client
    [...]
    error instanceof Prisma.PrismaClientKnownRequestError

**Reply by dennismoha** [reply]
Try this. Although I have a centralized error handler and logger.  just focus on the $on('error'..) 
```

`// lib/prismaClient.ts
import { PrismaClient, Prisma } from '@prisma/client';
import { config } from '@src/config';
import Logger from 'bunyan';
import { BadRequestError } from '@src/shared/globals/helpers/error-handler';


const logger: Logger = config.createLogger('prisma');

// Create a single instance of Prisma Client
const prismaClient = new PrismaClient({
    log: [
        {
            emit: 'event',
            level: 'query',
        },
        {
            emit: 'event',
            level: 'error',
        },
        {
            emit: 'event',
            level: 'info',
        },
        {
            emit: 'event',
            level: 'warn'
        }
    ],
});


// Log Prisma queries with the correct typing
prismaClient.$on('query', (e: Prisma.QueryEvent) => {    
    logger.debug(`Prisma Query: ${e.query} - ${JSON.stringify(e.params)}`);
});

// Log Prisma errors
prismaClient.$on('error', (e) => {
   
    // Handle Prisma known errors
    if (e instanceof Prisma.PrismaClientKnownRequestError) {
        if (e.code === 'P2002') {
          logger.error(
            `message: ${e.message}`
          );
        } else {
          logger.error(`Prisma Error: ${e.message}`);
        }
      } else {
        // Handle unexpected errors
        logger.error(`Unexpected error: ${e}`);
      }
    throw new BadRequestError('error');
    
});

// Log Prisma info events (optional)
prismaClient.$on('info', (e) => {
    logger.info(`Prisma Info: ${e.message}`);
});




// Optionally, you can handle disconnections and log errors in this file
prismaClient.$connect().catch((e) => {
    console.error('Prisma Client connection error:', e);
});

export default prismaClient;
`
```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!

**Comment by luunox** [comment]
use yarn

**Comment by mitchbart** [comment]
Instead of :
`import { Prisma } from "@prisma/client";`

try:
`import { Prisma } from '../../generated/prisma';`

make sure you update so the path to your generated folder is correct

**Reply by chaitanyashetty47** [reply]
thanks this worked, why is that the case tho? why isnt the "@prisma/client" import not working but ../../generated/prisma works tho?

**Reply by isSpicyCode** [reply]
thx, you’re the best! I’ve been trying to find the solution for 4 hours.

**Reply by Sourav1018** [reply]
> thx, you’re the best! I’ve been trying to find the solution for 4 hours.

True

**Comment by IanGraingerGMSL** [comment]
This doesn't work for me. prisma@6.18.0. `error instanceof Prisma.PrismaClientKnownRequestError` is false regardless how I import `Prisma`.

This does work:
```
    const prismaError = error as
      | { code?: string; constructor?: { name?: string } }
      | undefined;

    if (
      prismaError?.code === "P2002" &&
      prismaError?.constructor?.name === "PrismaClientKnownRequestError"
    ) {
```

**Comment by stouch** [comment]
> This doesn't work for me. prisma@6.18.0. error instanceof Prisma.PrismaClientKnownRequestError is false regardless how I import Prisma.

Same here with prisma@7.4.0


EDIT : 

Looks working with :

```typescript
import { Prisma } from '/path/to/database/prisma/generated/client';

// ...

  try {
    // ...
  } catch(err) {
    if (err instanceof Prisma.PrismaClientKnownRequestError && err.code === 'P2002') {
      // ...
    }
  }
```
